### PR TITLE
Add logicalTypes

### PR DESCRIPTION
**Pull request by kdvy**

### CODE REVIEW

**Comment by kdvy** [comment]
Iszak's had a look, if no one else has any issues I'll merge this and use this in our Avro code until (if) it gets merged into the main repo

File: logical_type.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+//////////////////////////////////////////////////////////////////////////////////////////////
+func timeMillisToNative(fn toNativeFn) toNativeFn {
+	return func(bytes []byte) (interface{}, []byte, error) {
+		l, b, err := fn(bytes)
+		if err != nil {
+			return l, b, err
+		}
+		i, ok := l.(int32)
+		if !ok {
+			return l, b, fmt.Errorf("cannot transform to native time.Duration, expected int, received %t", l)
```
Comment (JagGunawardana) [inline logical_type.go]: Should this be %F

File: logical_type.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+//////////////////////////////////////////////////////////////////////////////////////////////
+func timeMicrosToNative(fn toNativeFn) toNativeFn {
+	return func(bytes []byte) (interface{}, []byte, error) {
+		l, b, err := fn(bytes)
+		if err != nil {
+			return l, b, err
+		}
+		i, ok := l.(int64)
+		if !ok {
+			return l, b, fmt.Errorf("cannot transform to native time.Duration, expected long, received %t", l)
```
Comment (JagGunawardana) [inline logical_type.go]: As above

**Review by JagGunawardana — COMMENTED** [review]
Small comment on Printf flags - otherwise good.

**Pull request merged**